updater/lib/dependabot/api_client.rb

```diff
@@ -192,9 +193,9 @@ def fetch_job_details_from_backend
       sleep(rand(3.0..10.0)) && retry
     end
 
-    def create_pull_request_data(dependencies, updated_dependency_files, base_commit_sha, pr_message)
+    def create_pull_request_data(dependency_change, base_commit_sha)
```

Comment:
This method has gotten big enough I think it's probably worth extracting it into a serialiser class which presents a `dependency_change` for the API in a neater way.

I've punted on this for now as I want to land the concept of `DependencyChange` and I'm conscious it might change during code review.